File: go/raindrops/raindrops.go
package raindrops

import (
	"strconv"
)

const testVersion = 2

func Convert(n int) string {
	result := ""
	if n%3 == 0 {
		result = result + "Pling"
	}
	if n%5 == 0 {
		result = result + "Plang"
	}
	if n%7 == 0 {
		result = result + "Plong"
	}
	if result == "" {
		result = strconv.Itoa(n)
	}
	return result
}

File: haskell/raindrops/src/Raindrops.hs
module Raindrops (convert) where
import Data.List (intercalate)

convert :: Int -> String
convert n = 
  let divisions = [mapN d | d <- [3, 5, 7], mod n d == 0]
  in case divisions of
    [] -> show n
    _ -> intercalate "" divisions

mapN :: Int -> String
mapN 3 = "Pling"
mapN 5 = "Plang"
mapN 7 = "Plong"
